src/cue/contracts.cue

```diff
@@ -0,0 +1,76 @@
+package katapult
+
+contractBase: {
+  type?: string
+  slug?: string
+  data?: { ... }
+}
+
+ref :: contractBase & {
+  as?: string
+  cardinality?: string
+  version?: string
+}
+
+capabilityType: [Name=_]: contractBase & {
+  type: Name
+}
+
+capabilityType: endpoint: {
+  data: {
+    port: uint | *3000
+    protocol: "TCP" | "UDP" | *"TCP"
+  }
+}
+
+Contract :: contractBase & {
+  type: string
+  slug: string
+
+  requires?: [...ref]
+  provides?: [...ref]
+
+  // Default capabilities inferred from the type.
+  if (type == "sw.containerized-scalable-service" || type == "sw.containerized-service") {
+    provides: [...ref] | *[capabilityType.endpoint & { as: "main-endpoint" }]
```

Comment:
This way we define that services provide an endpoint by default (even if nothing is specified in `provides`).